profiles/repositories.py

```diff
@@ -89,8 +93,22 @@ def generate_id():
         self.db = db
         self._next_id_generator = next_id_generator
 
-    def add(self, profile: Profile) -> None:
+    def add(self, profile: Profile) -> Profile:
+        self.db.session.begin_nested()
         self.db.session.add(profile)
+        try:
+            self.db.session.commit()
+        except IntegrityError as exception:
+            self.db.session.rollback()
+
+            if profile.orcid:
+                LOGGER.info('Profile for ORCID %s appears to already exist', profile.orcid)
```

Comment:
assumes ORCID is the only field that can cause an `IntegrityError`? A more conservative message could be `Cannot insert profile, assuming its existence to retrieve it` or similar

Reply:
Could be email address too. Will expand.